tasks/upload_processor.py

```diff
@@ -436,77 +450,65 @@ def should_delete_archive(self, commit_yaml: UserYaml) -> bool:
-                    extra=dict(
-                        archive_url=archive_url,
-                        commit=commit.commitid,
-                        upload=upload.external_id,
-                        parent_task=self.request.parent_id,
-                    ),
-                )
-                archive_service = report_service.get_archive_service(commit.repository)
-                archive_service.delete_file(archive_url)
-                return True
-        return False
+    @sentry_sdk.trace
+    def postprocess_raw_reports(
+        self,
+        report_service: ReportService,
+        commit: Commit,
+        reports: list[RawReportInfo],
+    ):
+        should_delete_archive = self.should_delete_archive(report_service.current_yaml)
+        archive_service = report_service.get_archive_service(commit.repository)
 
-    def _attempt_rewrite_raw_report_readable_error_case(
-        self, report_service: ReportService, commit: Commit, upload: Upload
-    ) -> None:
-        log.info(
-            "Attempting to rewrite raw upload in readable format for debugging purposes (processing already failed)",
-            extra=dict(commit=commit.commitid, upload=upload.external_id),
-        )
-        try:
-            raw_report = report_service.parse_raw_report_from_storage(
-                commit.repository, upload, is_error_case=True
-            )
-            self._rewrite_raw_report_readable(
-                raw_report,
-                upload,
-                report_service=report_service,
-                commit=commit,
-            )
-        except FileNotFoundError:
-            log.exception(
-                "Failed to rewrite raw report in readable format",
-                extra=dict(commit=commit.commitid, upload=upload.external_id),
-            )
+        if should_delete_archive:
+            for report_info in reports:
+                archive_url = report_info.archive_url
+                if not archive_url.startswith("http"):
+                    log.info(
+                        "Deleting uploaded file as requested",
+                        extra=dict(
+                            archive_url=archive_url,
+                            commit=commit.commitid,
+                            upload=report_info.upload,
+                            parent_task=self.request.parent_id,
+                        ),
+                    )
+                    archive_service.delete_file(archive_url)
+
+        else:
+            for report_info in reports:
+                self._rewrite_raw_report_readable(archive_service, commit, report_info)
 
     def _rewrite_raw_report_readable(
         self,
-        raw_report: ParsedRawReport,
-        upload: Upload,
-        report_service: ReportService,
+        archive_service: ArchiveService,
         commit: Commit,
+        raw_report_info: RawReportInfo,
     ) -> None:
-        archive_url = upload.storage_path
+        assert raw_report_info.raw_report
```

Comment:
Hum... this could throw an exception, preventing any reports that would come after this one in the loop from being re-written.

Are we fine with that?

Reply:
This is a bit unfortunate. I use `RawReportInfo` as an "out-parameter", and because of typing, `raw_report` can be `None`.

However `_rewrite_raw_report_readable` is only ever called in code paths that check that `raw_report` is in fact not `None`, and to satisfy types within that function, I assert that fact inside the function, but mostly only for the type checker.

I understand this is quite unfortunate, and ideally I would like to get rid of all the error handling that made me turn this into an out-parameter. I will take another look to figure out if I can solve things a bit differently.